apps/src/blockly/customBlocks/cdoBlockly/spritelabBlocks.js

```diff
@@ -279,4 +280,94 @@ export const blocks = {
     };
     generator.sprite_parameter_get = generator.variables_get;
   },
+
+  // All logic for behavior picker custom input type
+  behaviorPicker() {
+    return {
+      addInput(blockly, block, inputConfig, currentInputRow) {
+        currentInputRow
+          .appendField(inputConfig.label)
+          .appendField(
+            new Blockly.FieldDropdown(getAllBehaviors, undefined, undefined),
+            inputConfig.name
+          );
+        let allowBehaviorEditing = Blockly.useModalFunctionEditor;
+        // Don't allow behaviors to be edited for uncategorized toolboxes or read only workspaces.
```

Comment:
I know this is copied over and seems to work now, but the logic seems wonky here. We don't allow behavior editing if we have app options, if the level has a toolbox, is not a readonly workspace, and it does not have categories. Should we be doing a check for does not have categories OR is a read only workspace? If we think it's a problem we could file a follow-up task or just decide to keep it as-is since we are retiring this version soon and haven't heard complaints.

Reply:
Personally, I’d vote for keeping this as-is unless we suddenly hear complaints. Theoretically we should be able to delete all this code in a couple months, so I’m not sure I see the value in tweaking it now.